pydra/engine/specs.py

```diff
@@ -449,6 +449,10 @@ def _field_defaultvalue(self, fld, output_dir):
 
     def _field_metadata(self, fld, inputs, output_dir):
         """Collect output file if metadata specified."""
+        if "requires" in fld.metadata:
+            for inp in fld.metadata["requires"]:
+                if getattr(inputs, inp) in [attr.NOTHING, None, False]:
```

Comment:
`False` could be a valid state for a field. 

```suggestion
                if getattr(inputs, inp) in [attr.NOTHING, None]:
```

Reply:
but usually you use flag that are bool